packages/kbn-esql-utils/src/utils/run_query_utils.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { DatatableColumn } from '@kbn/expressions-plugin/common';
+import type { ISearchStart } from '@kbn/data-plugin/public';
+import { esFieldTypeToKibanaFieldType } from '@kbn/field-types';
+import type { ESQLSearchReponse } from '@kbn/es-types';
+import { lastValueFrom } from 'rxjs';
+import { ESQL_LATEST_VERSION } from '../../constants';
+
+export async function getESQLQueryColumns({
+  esqlQuery,
+  search,
+  signal,
+}: {
+  esqlQuery: string;
+  search: ISearchStart;
+  signal?: AbortSignal;
+}): Promise<DatatableColumn[]> {
+  const response = await lastValueFrom(
+    search.search(
+      {
+        params: {
+          query: `${esqlQuery} | limit 0`,
```

Comment:
I forgot to add the limit 0 as @dej611 correctly noticed. 

@dej611 good catch!